peer.go

```diff
@@ -399,41 +397,43 @@ func (p *Peer) AddOutboundConnection(c *Connection) error {
 	p.Lock()
 	p.outboundConnections = append(p.outboundConnections, c)
 	p.Unlock()
-
-	p.connectionStateChanged(c)
 	return nil
 }
 
-// checkInboundConnection will check whether the changed connection is an inbound
-// connection, and will remove any closed connections.
-func (p *Peer) checkInboundConnection(changed *Connection) (updated bool, isInbound bool) {
-	newConns := p.inboundConnections[:0]
-	for _, c := range p.inboundConnections {
+// removeClosed will check remove the changed connection if it exists
+// in the specified connections list.
+func (p *Peer) removeClosed(connsPtr *[]*Connection, changed *Connection) (updated bool, found bool) {
+	conns := *connsPtr
+	newConns := conns[:0]
+	for _, c := range conns {
 		if c == changed {
-			isInbound = true
+			found = true
 		}
 
-		if c.readState() != connectionClosed {
-			newConns = append(newConns, c)
-		} else {
-			updated = true
-		}
+		updated = true
 	}
 	if updated {
-		p.inboundConnections = newConns
+		*connsPtr = newConns
 	}
 
-	return updated, isInbound
+	return updated, found
```

Comment:
I think I'm missing something here - don't we need to append the non-matching conns to `newConns` somewhere?